beaker/src/depfile_generator.rs

```diff
@@ -0,0 +1,285 @@
+//! Make-compatible dependency file generation
```

Comment:
I'm concerned the output tracking will get out of sync with the actual outputs produced. Can we have OutputManager keep track of the outputs actually produced by a subcommand and then use it to produce the dep file? We can keep the helper functions here for dep file syntax but I would like the outputs to be driven by what OutputManager actually saves, rather than rederived here